src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -147,18 +151,31 @@ export class EditorPart extends Part implements IEditorPart {
 		@IThemeService themeService: IThemeService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IStorageService storageService: IStorageService,
-		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService
+		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService,
 	) {
 		super(id, { hasTitle: false }, themeService, storageService, layoutService);
 
 		this.registerListeners();
 	}
 
 	private registerListeners(): void {
+		this._register(this.onDidSetGridWidget(() => {
```

Comment:
I find it a bit strange that this listener is only installed after the grid widget is set. Why the coupling?